plugins/aea-test-autonomy/aea_test_autonomy/fixture_helpers.py

```diff
@@ -584,6 +584,23 @@ def _build_image(cls) -> DockerImage:
         return AMMNetDockerImage(client, cls.addr, cls.port)
 
 
+class RegistriesBaseTest(HardHatBaseTest):
+    """Base pytest class for component registries."""
+
+    addr: str = DEFAULT_HARDHAT_ADDR
+    port: int = DEFAULT_HARDHAT_PORT
+
+    @classmethod
+    def _build_image(cls) -> DockerImage:
+        """Build the image."""
+        client = docker.from_env()
+        return RegistriesDockerImage(
+            client=client,
+            addr=cls.addr,
+            port=cls.port,
+        )
+
+
```

Comment:
What does this accomplish that `UseRegistries` doesn't?

Reply:
I was not aware of the base class, I tried searching for base test class but couldn't find one. Now I know why, I'll update the code

Reply:
I tried using the `UseRegistries` but there is an issue, it's not designed for testing the contracts. The  `HardHatBaseTest ` and all the classes derived from it are compatible with the `BaseContractTest`.